### ISSUE TITLE
Filter before open file?

### ISSUE THREAD

**Issue by ankkk** [pascalbaljetmedia/laravel-ffmpeg#139]
I have one static image (JPEG), me need to get 15 seconds of video. I do it like this:
```
FFMpeg::fromDisk('temp')
    ->open('test.jpeg')
    ->addFilter('-loop', 1)
    ->addFilter('-t', 15)
    ->addFilter('-r', 10)
    ->addFilter('-vf', 'pad=ceil(iw/2)*2:ceil(ih/2)*2')
    ->addFilter('-pix_fmt', 'yuv420p')
    ->addFilter('-profile:v', 'baseline', '-level', 3)
    ->addFilter('-crf', 18)
    ->export()
    ->toDisk('movies')
    ->inFormat(
        (new X264())->setKiloBitrate(1500)
    )
    ->save('test.mp4');
```

But I get 1 second video because `-loop 1` is substituted after opening the image, like this:
```
'/bin/ffmpeg' '-y' '-i' '/var/www/storage/app/temp/test.jpeg' '-loop' '1' '-t' '15' '-r' '10' '-pix_fmt' 'yuv420p' '-profile:v' 'baseline' '-level' '3' '-crf' '18' '-threads' '12' '-vcodec' 'libx264' '-acodec' 'aac' '-b:v' '3000k' '-refs' '6' '-coder' '1' '-sc_threshold' '40' '-flags' '+loop' '-me_range' '16' '-subq' '7' '-i_qfactor' '0.71' '-qcomp' '0.6' '-qdiff' '4' '-trellis' '1' '-b:a' '128k' '-vf' '[in]pad=ceil(iw/2)*2:ceil(ih/2)*2[out]' '-pass' '2' '-passlogfile' '/tmp/ffmpeg-passes5d11aa6a97550hgfxa/pass-5d11aa6a97649' '/var/www/storage/app/movies/test.mp4' 
```

How can I fix this, or is there some other way to get a video of the desired length?

**Comment by jhoysner** [comment]
friend you managed to solve it, I still can't get it to last more than 1s, and sometimes I get an error if I put other filters

FFMpeg\Exception\RuntimeException
Encoding failed

**Comment by pascalbaljet** [comment]
I've rewritten the package and support for timelapses is included. Could you test the `v7` branch? Almost no breaking changes except for some filter stuff in HLS exports.

https://github.com/pascalbaljetmedia/laravel-ffmpeg/tree/v7

`composer require "pbmedia/laravel-ffmpeg:v7.x-dev"`

```php
<?php

FFMpeg::open('test.jpg')
    ->export()
    ->asTimelapseWithFramerate(15)
    ->inFormat(new X264)
    ->save('timelapse.mp4');
```

**Comment by pascalbaljet** [comment]
Fixed in `v7.0`
